straxen/plugins/event_w_bayes_class.py

```diff
@@ -0,0 +1,55 @@
+import strax
+import straxen
+import numpy as np
+
+
+class EventwBayesClass(strax.Plugin):
+    """
+    Append at event level the posterior probability for an
+    S1, S2, alt_S1 and alt_S2
+    """
+    provides = 'event_w_bayes_class'
+    depends_on = ('peaklet_classification_bayes', 'event_basics')
+    data_kind = 'events'
+    __version__ = '0.0.1'
+
+    def infer_dtype(self):
+
+        dtype = []
+        dtype += strax.time_fields
+        for name in ['s1', 's2', 'alt_s1', 'alt_s2']:
+            dtype += [(f'{name}_s1_ln_prob', np.float32, f'Given an {name}, s1 ln probability')]
+            dtype += [(f'{name}_s2_ln_prob', np.float32, f'Given an {name}, s2 ln probability')]
+
+        return dtype
+
+    def compute(self, peaklets, events):
+
+        result = np.empty(len(events), dtype=self.dtype)  # caution here zeros means 1 remember this is ln(Pr), so do not initiate with zeros
+
+        s1 = np.zeros(len(events), strax.time_fields)
+        s2 = np.zeros(len(events), strax.time_fields)
+        s1_alt = np.zeros(len(events), strax.time_fields)
+        s2_alt = np.zeros(len(events), strax.time_fields)
+
+        # For S1 and S2, match to peak posteriors
+        for peaks, name in zip([s1, s2, s1_alt, s2_alt], ['s1', 's2', 'alt_s1', 'alt_s2']):
+            peaks['time'] = events[f'{name}_time']
+            peaks['endtime'] = events[f'{name}_endtime']
+            fci = strax.fully_contained_in(peaklets, peaks)
```

Comment:
If I remember correctly, you can't use this function as peaklets aren't sorted by time since an S1 can happen during an S2

Reply:
n/a